iothub/device/src/IotHubConnection.cs

```diff
@@ -110,14 +110,10 @@ protected IotHubConnection(string hostName, int port, AmqpTransportSettings amqp
                         break;
                 }
 
-                SetLinkSettingsCommonProperties(linkSettings, timeoutHelper.RemainingTime(), productInfo);
-                if (linkType == SendingLinkType.Methods)
+                SetLinkSettingsCommonProperties(linkSettings, timeoutHelper.RemainingTime(), productInfo, corrId, linkType.ToString());
```

Comment:
Please avoid using linkType.ToString() (for both perf issues and separation of concerns - we should be able to change the internal enum name without affecting bytes on the wire).

Reply:
Sure thing, I'll change this